epsagon_kubernetes_subscription.sh

```diff
@@ -0,0 +1,105 @@
+    echo "Fetching epsagon-role.yaml"
+    if [ -f 'epsagon-role.yaml' ] ; then
+        echo "epsagon-role.yaml already exists - using that file"
+        return 0
+    fi
+    if [ `which wget` ] ; then
+        wget $ROLE_URL
+    else
+        if [ `which curl` ] ; then
+            curl $ROLE_URL -o epsagon-role.yaml
+        else
+            if [ -z epsagon-role.yaml ] ; then
+                echo "Could not get epsagon-role.yaml"
+                echo "Please download the role from:"
+                echo $ROLE_URL
+                exit 1
+            fi
+        fi
+    fi
+}
+
+function send_to_epsagon {
+    EPSAGON_TOKEN=$1
+    ROLE_TOKEN=$2
+    if [ $# == 3 ]; then
+        CONTEXT=$3
+        SERVER=`kubectl config view | grep -B 3 -E "[[:space:]]$CONTEXT\>" | grep -E "\<server: " | awk '{print $2}'`
+    else
+        # no context
+        echo "TODO"
+    fi
+    if [ `which curl` ] ; then
+        curl -X POST https://api.epsagon.com/containers/k8s/add_cluster_by_token -d "{\"k8s_cluster_url\": \"$SERVER\", \"epsagon_token\": \"$EPSAGON_TOKEN\", \"cluster_token\": \"$ROLE_TOKEN\"}" -H 'Content-Type: application/json'
+        echo ""
+    else
+        echo "Could not find 'curl' command to send data to epsagon, please enter manually"
+        echo "server=${SERVER}"
+
+        echo ""
+        echo "--------"
+        echo ""
+        echo "The token for the epsagon role is: "
+        echo ""
+        echo $ROLE_TOKEN
+        echo ""
+    fi
+}
+
+function apply_role {
+    EPSAGON_TOKEN=$1
+    KUBECTL="kubectl"
+    if [ ! -z $2 ] ; then
+        CONTEXT=$2
+        KUBECTL="kubectl --context ${CONTEXT}"
+        echo "Applying epsagon-role.yaml to ${CONTEXT}"
+    else
+        echo "Applying epsagon-role.yaml"
+    fi
+    echo ""
+    ${KUBECTL} apply -f epsagon-role.yaml
+
+    SA_SECRET_NAME=`${KUBECTL} -n epsagon-monitoring get secrets | grep 'epsagon-monitoring-token' | awk '{print $1}'`
+    if [ `which python` ] ; then
+        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json | python -c 'import sys, json; print(json.load(sys.stdin)["data"]["token"])' | base64 -d`
+    else
+        ROLE_TOKEN=`${KUBECTL} -n epsagon-monitoring get secrets $SA_SECRET_NAME -o json | grep '\"token\"' | cut -d: -f2 | cut -d'"' -f2 | base64 -d`
+    fi
+
+    send_to_epsagon $EPSAGON_TOKEN $ROLE_TOKEN $CONTEXT
+
+}
+
+function apply_epsagon_on_all_contexts {
+    echo "Welcome to Epsagon!"
+    NUMBER_OF_CONTEXTS=$((`kubectl config get-contexts | wc -l` - 1 ))
+    for context in `kubectl config get-contexts | tail -${NUMBER_OF_CONTEXTS} | sed s/^*//  | awk {'print $1'}`; do
+        echo ""
+        echo "Now installing Epsagon to: $context"
+        echo -n "Would you like to proceed? [Y/N] "
+        read answer
+        if [ "$answer" == "Y" ]; then
+            apply_role $1 $context
+        else
+            continue
+        fi
+    done
+}
+
+if [ $# -ne 1 ] ; then
```

Comment:
Can we validate other things regarding the token (match a regex, or length)?

Reply:
There are million things to check, but ultimately if someone enters the wrong token they will get an error later. I don't think we need to verify more things, it will just complicate things here.